### ISSUE TITLE
Skipped tests that generate warnings

### ISSUE THREAD

**Issue by nabobalis** [sunpy/sunpy#4670]
I found out in #4358 that when we ran only online tests, this meant that no doctests were collected anywhere.
This change was reverted and several doctests were fixed or skipped in order to prevent the CI from breaking.

While this will not be a full list, some of the changes made that need to be updated/fixed:

```
____________________________________________________________________ [doctest] sunpy.map.map_factory.MapFactory ____________________________________________________________________
095     `sunpy.io.header.FileHeader` or `sunpy.util.metadata.MetaDict` classes.
096 
097     * data, header pairs, not in tuples
098 
099     >>> mymap = sunpy.map.Map(data, header)   # doctest: +SKIP
100 
101     * data, wcs object, in tuple
102 
103     >>> from astropy.wcs import WCS
104     >>> wcs = WCS(sunpy.data.sample.AIA_171_ROLL_IMAGE)     # doctest: +REMOTE_DATA +IGNORE_WARNINGS
UNEXPECTED EXCEPTION: VerifyWarning('Verification reported errors:')
Traceback (most recent call last):
  File "/opt/miniconda/envs/sunpy/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest sunpy.map.map_factory.MapFactory[7]>", line 1, in <module>
  File "/opt/miniconda/envs/sunpy/lib/python3.8/site-packages/astropy/wcs/wcs.py", line 425, in __init__
    header_string = header.tostring().rstrip()
  File "/opt/miniconda/envs/sunpy/lib/python3.8/site-packages/astropy/io/fits/header.py", line 684, in tostring
    s = str(card)
  File "/opt/miniconda/envs/sunpy/lib/python3.8/site-packages/astropy/io/fits/card.py", line 205, in __str__
    return self.image
  File "/opt/miniconda/envs/sunpy/lib/python3.8/site-packages/astropy/io/fits/card.py", line 521, in image
    self.verify('fix+warn')
  File "/opt/miniconda/envs/sunpy/lib/python3.8/site-packages/astropy/io/fits/verify.py", line 117, in verify
    warnings.warn(line, VerifyWarning)
astropy.io.fits.verify.VerifyWarning: Verification reported errors:
```

```
_____________________________________________________________ [doctest] sunpy.timeseries.sources.eve.EVESpWxTimeSeries _____________________________________________________________
153     For other levels of EVE data, use `~sunpy.net.Fido`, with `sunpy.net.attrs.Instrument('eve')`.
154 
155     Data is available starting on 2010/03/01.
156 
157     Examples
158     --------
159     >>> import sunpy.timeseries
160     >>> import sunpy.data.sample  # doctest: +REMOTE_DATA
161     >>> eve = sunpy.timeseries.TimeSeries(sunpy.data.sample.EVE_TIMESERIES, source='EVE')  # doctest: +REMOTE_DATA
162     >>> eve = sunpy.timeseries.TimeSeries("http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/LATEST_EVE_L0CS_DIODES_1m.txt", source='EVE')  # doctest: +REMOTE_DATA
UNEXPECTED EXCEPTION: SunpyUserWarning('Unknown units for x_cool proxy.')
Traceback (most recent call last):
  File "/opt/miniconda/envs/sunpy/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest sunpy.timeseries.sources.eve.EVESpWxTimeSeries[3]>", line 1, in <module>
  File "/home/nabil/GitHub/sunpy/sunpy/timeseries/timeseries_factory.py", line 481, in __call__
    timeseries._sanitize_units()
  File "/home/nabil/GitHub/sunpy/sunpy/timeseries/timeseriesbase.py", line 565, in _sanitize_units
    warnings.warn(f"Unknown units for {column}.", SunpyUserWarning)
sunpy.util.exceptions.SunpyUserWarning: Unknown units for x_cool proxy.
```

```
______________________________________________________ [doctest] sunpy.physics.differential_rotation.solar_rotate_coordinate _______________________________________________________
227     >>> import astropy.units as u
228     >>> from astropy.coordinates import SkyCoord, get_body
229     >>> from sunpy.coordinates import Helioprojective
230     >>> from sunpy.physics.differential_rotation import solar_rotate_coordinate
231     >>> from sunpy.time import parse_time
232     >>> start_time = parse_time('2010-09-10 12:34:56')
233     >>> end_time = parse_time('2010-09-11 13:34:56')
234     >>> c = SkyCoord(-570*u.arcsec, 120*u.arcsec, obstime=start_time,
235     ...              observer="earth", frame=Helioprojective)
236     >>> solar_rotate_coordinate(c, time=end_time)  # doctest: +IGNORE_WARNINGS
UNEXPECTED EXCEPTION: UserWarning("Using 'time' assumes an Earth-based observer.")
Traceback (most recent call last):
  File "/opt/miniconda/envs/sunpy/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest sunpy.physics.differential_rotation.solar_rotate_coordinate[8]>", line 1, in <module>
  File "/home/nabil/GitHub/sunpy/sunpy/physics/differential_rotation.py", line 247, in solar_rotate_coordinate
    new_observer = _get_new_observer(coordinate.obstime, observer, time)
  File "/home/nabil/GitHub/sunpy/sunpy/physics/differential_rotation.py", line 168, in _get_new_observer
    warnings.warn("Using 'time' assumes an Earth-based observer.")
UserWarning: Using 'time' assumes an Earth-based observer.
/home/nabil/GitHub/sunpy/sunpy/physics/differential_rotation.py:236: UnexpectedException
```

Furthermore:
```
    # Some tests use data that have dates in the future and ERFA does not like.
    ignore:ERFA function "d2dtf"*
    ignore:ERFA function "dtf2d"*
    # Currently helio makes unverified requests - this filter should be removed when
    # https://github.com/sunpy/sunpy/issues/4401 is fixed
    ignore:Unverified HTTPS request
```
was added to the setup.cfg to cover up more warnings.

In docs/guide/data_types/maps.rst at the end of the file:
```
    >>> from sunpy.physics.solar_rotation import calculate_solar_rotate_shift
    >>> shifts = calculate_solar_rotate_shift(mc)  # doctest: +SKIP
```
was skipped due to further errors.

**Comment by ayshih** [comment]
> I found out in #4358 that when we ran only online tests, this meant that no doctests were collected anywhere.

Ah, that explains why some doctests I thought might fail weren't failing in CI.  I just figured that doctests were still passing due the tolerance of FLOAT_CMP.  In retrospect, it seems "obvious" that filtering by marker won't still run all of the doctests.

**Comment by nabobalis** [comment]
> > I found out in #4358 that when we ran only online tests, this meant that no doctests were collected anywhere.
> 
> Ah, that explains why some doctests I thought might fail weren't failing in CI. I just figured that doctests were still passing due the tolerance of FLOAT_CMP. In retrospect, it seems "obvious" that filtering by marker won't still run all of the doctests.

Yeah I feel like the world's biggest idiot.

**Comment by kakirastern** [comment]
But you are definitely not from what I observe

**Comment by ayshih** [comment]
Two of these failures are odd because they are erroring due to warnings that should be ignored due to the `+IGNORE_WARNINGS` flag.  For example, the warning for the `solar_rotate_coordinate()` doctest is correctly ignored locally if I use `+IGNORE_WARNINGS`.  Perhaps there was something in the testing configuration at the time that was overriding `+IGNORE_WARNINGS`?

**Comment by ayshih** [comment]
`pytest-doctestplus` has a bug that prevented the use of `+IGNORE_WARNINGS` in our `online` build.  I have PR-ed a fix  here: astropy/pytest-doctestplus#152
